ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,245 @@
+    """Run a command in the OpenShift cluster."""
+    result = subprocess.run(
+        ["oc", *args],  # noqa: S603, S607
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+    return result
+
+
+def get_cluster_id():
+    """Get the cluster ID."""
+    try:
+        result = run_oc(
+            ["get", "clusterversion", "-o", "jsonpath='{.items[].spec.clusterID}'"]
+        )
+        cluster_id = result.stdout.replace("'", "")
+        return cluster_id
+    except subprocess.CalledProcessError as e:
+        raise Exception("Error getting cluster ID") from e
+
+
+def get_cloud_openshift_pull_secret():
+    """Get the cloud.openshift.com pull secret."""
+    try:
+        result = run_oc(
+            [
+                "get",
+                "secret",
+                "pull-secret",
+                "-n",
+                "openshift-config",
+                "-o",
+                r"jsonpath='{.data.\.dockerconfigjson}'",
+            ]
+        )
+        pull_secret = result.stdout
+        # decode the base64-encoded dockerconfigjson
+        dockerconfigjson = base64.b64decode(pull_secret).decode("utf-8")
+        # parse the JSON data
+        dockerconfig = json.loads(dockerconfigjson)
+        # extract the auth value
+        auth = dockerconfig["auths"]["cloud.openshift.com"]["auth"]
+        return auth
+    except subprocess.CalledProcessError as e:
+        raise Exception("Error getting pull secret") from e
+
+
+def collect_ols_data_from(location: str) -> list[pathlib.Path]:
+    """Collect files from a given location.
+
+    Args:
+        location: Path to the directory to be searched for files.
+
+    Returns:
+        List of paths to the collected files.
+
+    Only JSON files from the 'feedback' and 'transcripts' directories are collected.
+    """
+    files = []
+
+    files += list(pathlib.Path(location).glob("feedback/*.json"))
+    files += list(pathlib.Path(location).glob("transcripts/*/*/*.json"))
+
+    return files
+
+
+def package_files_into_tarball(
+    file_paths: list[pathlib.Path], path_to_strip: str
+) -> io.BytesIO:
+    """Package specified directory into a tarball.
+
+    Args:
+        file_paths: List of paths to the files to be packaged.
+        path_to_strip: Path to be stripped from the file paths (not
+            included in the archive).
+
+    Returns:
+        BytesIO object representing the tarball.
+    """
+    tarball_io = io.BytesIO()
+
+    with tarfile.open(fileobj=tarball_io, mode="w:gz") as tar:
+        # arcname parameter is set to a stripped path to avoid including
+        # the full path of the root dir
+        for file_path in file_paths:
+            tar.add(file_path, arcname=file_path.as_posix().replace(path_to_strip, ""))
+
+        # add magic file for identification of our archive on the CCX side
+        empty_file = tarfile.TarInfo("openshift_lightspeed.json")
+        empty_file.size = 0
+        tar.addfile(empty_file)
```

Comment:
is there any size limit on the tar file insights will accept? 

even if there's no limit enforced by insights, I think we should put a constraint on the max size and create/upload multiple files when we hit that size limit (say 10 megs or something)

Reply:
another story please: https://issues.redhat.com/browse/OLS-473

Reply:
sure